diff --git a/ckan/views/user.py b/ckan/views/user.py
index 00efe36b7..2efc7614c 100644
--- a/ckan/views/user.py
+++ b/ckan/views/user.py
@@ -508,26 +508,23 @@ class RequestResetView(MethodView):
         context, data_dict = self._prepare()
         id = data_dict[u'id']
 
-        context = {u'model': model, u'user': g.user}
         user_obj = None
         try:
-            logic.get_action(u'user_show')(context, data_dict)
-            user_obj = context[u'user_obj']
+            user_obj = model.User.get(id)
+            if user_obj is None:
+                raise logic.NotFound
         except logic.NotFound:
             # Try searching the user
             if id and len(id) > 2:
-                user_list = logic.get_action(u'user_list')(context, {
-                    u'id': id
-                })
+                user_list = model.Session.query(model.User.name).all()
                 if len(user_list) == 1:
                     # This is ugly, but we need the user object for the
                     # mailer,
                     # and user_list does not return them
                     data_dict[u'id'] = user_list[0][u'id']
-                    logic.get_action(u'user_show')(context, data_dict)
-                    user_obj = context[u'user_obj']
+                    user_obj = model.User.get(id)
                 elif len(user_list) > 1:
-                    h.flash_error(_(u'"%s" matched several users') % (id))
+                    h.flash_error(_(u'No such user: %s') % (id))
                 else:
                     h.flash_error(_(u'No such user: %s') % id)
             else:
