diff --git a/ckan/public/base/javascript/modules/resource-reorder.js b/ckan/public/base/javascript/modules/resource-reorder.js
index 789b123..02caf15 100644
--- a/ckan/public/base/javascript/modules/resource-reorder.js
+++ b/ckan/public/base/javascript/modules/resource-reorder.js
@@ -4,7 +4,7 @@ this.ckan.module('resource-reorder', function($) {
   return {
     options: {
       id: false,
-      labelText: 'Reorder resources'
+      labelText: 'Reorder'
     },
     template: {
       title: '<h1></h1>',
@@ -110,9 +110,17 @@ this.ckan.module('resource-reorder', function($) {
         module.html_saving.show();
         $('.save, .cancel', module.html_form_actions).addClass('disabled');
         var order = [];
+        var pinned = null;
         $('.resource-item', module.el).each(function() {
-          order.push($(this).data('id'));
+          if ($(this).data('pinned') === 'True') {
+            pinned = $(this).data('id')
+          } else {
+            order.push($(this).data('id'));
+          }
         });
+        if (pinned) {
+          order.unshift(pinned)
+        }
         module.sandbox.client.call('POST', 'package_resource_reorder', {
           id: module.options.id,
           order: order

