diff --git a/ckan/public/base/javascript/modules/resource-reorder.js b/ckan/public/base/javascript/modules/resource-reorder.js
index 3489742..5426ffb 100644
--- a/ckan/public/base/javascript/modules/resource-reorder.js
+++ b/ckan/public/base/javascript/modules/resource-reorder.js
@@ -38,7 +38,7 @@ this.ckan.module('resource-reorder', function($) {
     initialize: function() {
       jQuery.proxyAll(this, /_on/);
 
-      var labelText = this._('Reorder resources');
+      var labelText = this._('Reorder');
       var helpText = this._('You can rearrange the resources by dragging them using the arrow icon. Drag the resource ' +
         'to the right and place it to the desired location on the list. When you are done, click the "Save order" -button.');
 
@@ -119,9 +119,17 @@ this.ckan.module('resource-reorder', function($) {
         module.html_saving.show();
         $('.save, .cancel', module.html_form_actions).addClass('disabled');
         var order = [];
+        var pinned = null;
         $('.resource-item', module.el).each(function() {
-          order.push($(this).data('id'));
+          if ($(this).data('pinned') === 'True') {
+            pinned = $(this).data('id')
+          } else {
+            order.push($(this).data('id'));
+          }
         });
+        if (pinned) {
+          order.unshift(pinned)
+        }
         module.sandbox.client.call('POST', 'package_resource_reorder', {
           id: module.options.id,
           order: order
