diff --git a/ckan/lib/uploader.py b/ckan/lib/uploader.py
index 146e1fe28..47bc82d23 100644
--- a/ckan/lib/uploader.py
+++ b/ckan/lib/uploader.py
@@ -219,7 +219,7 @@ class ResourceUpload(object):
         if url and config_mimetype_guess == 'file_ext' and urlparse(url).path:
             self.mimetype = mimetypes.guess_type(url)[0]
 
-        if bool(upload_field_storage) and \
+        if upload_field_storage is not None and \
                 isinstance(upload_field_storage, ALLOWED_UPLOAD_TYPES):
             self.filesize = 0  # bytes
 
